server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -369,8 +470,8 @@ private static HitContext prepareNestedHitContext(
         assert nestedIdentity != null;
         Source nestedSource = nestedIdentity.extractSource(rootSource);
 
-        SearchHit hit = new SearchHit(topDocId, rootId, nestedIdentity);
-        return new HitContext(hit, subReaderContext, nestedInfo.doc(), childFieldLoader.storedFields(), nestedSource, rankDoc);
+        SearchHit nestedHit = new SearchHit(topDocId, rootId, nestedIdentity);
```

Comment:
NIT: unnecessary rename noise :)

Reply:
I think it clarifies (for me at least) which hit we're dealing with.